### PR TITLE
Set error if malloc returns NULL

### PR DESCRIPTION
**Pull request by richsalz**
This is a partial fix for #5841.  I hope @FdaSilvaYY will finish off all the x509 onces as part of his PR #5837.

I am not going to backport:

-  Malloc failures will often result in program crashes anyway.
- The error stuff is very different for 1.1.1
- 1.0.2 is almost out of LTS lifetime
- Few will stay on 1.1.0 when 1.1.1 and TLS 1.3 comes out
- blah blah blah

### CODE REVIEW

**Comment by mspncp** [comment]
I like your last argument :wink:.

**Comment by mspncp** [comment]
You will squash this all into a single commit, right?

File: crypto/stack/stack.c [hunk 1]
```diff
@@ -173,9 +173,10 @@ static int sk_reserve(OPENSSL_STACK *st, int n, int exact)
          * At this point, |st->num_alloc| and |st->num| are 0;
          * so |num_alloc| value is |n| or |min_nodes| if greater than |n|.
          */
-        st->data = OPENSSL_zalloc(sizeof(void *) * num_alloc);
-        if (st->data == NULL)
+        if ((st->data = OPENSSL_zalloc(sizeof(void *) * num_alloc)) == NULL) {
+            /* STACKerr(STACK_F_SK_RESERVE, ERR_R_MALLOC_FAILURE); */
```
Comment (FdaSilvaYY) [inline crypto/stack/stack.c]: Errors from stack are reported by caller and stack user...

Comment (richsalz) [reply crypto/stack/stack.c]: I don't understand this comment?

**Comment by richsalz** [comment]
Yes I will squash it before merging.  Will wait a bit for @FdaSilvaYY to clarify his comment tho.

**Comment by richsalz** [comment]
Thanks for boring tolerance :)

File: crypto/stack/stack.c (hunk 1 above)
Comment (FdaSilvaYY) [reply crypto/stack/stack.c]: I say that this code is useless: 
it has no value added, it just says that one of  the many stack is KO, ... and please guess which one  ;) 
Actually, the current usage is to report this kind of failure just from the caller code, just after the failed call .

Comment (richsalz) [reply crypto/stack/stack.c]: No, fine-grain "what went wrong" and then pushing more errors as we pop up the stack is the  way we do things.

**Pull request closed without merging**